File: app/src/main/java/ru/ulstu_team/ulstuschedule/util/FontUtil.java
package ru.ulstu_team.ulstuschedule.util;

import android.content.Context;
import android.graphics.Typeface;
import android.util.Log;

public class FontUtil {

    private static final String TAG = "FontUtil";

    private FontUtil() {}

    public static Typeface getFont(Context context, String fontName) {
        try {
            String pathBegin = "fonts/";
            if (!fontName.startsWith(pathBegin))
                fontName = pathBegin + fontName;
            return Typeface.createFromAsset(context.getAssets(), fontName);
        } catch (Exception ex) {
            Log.e(TAG, "getFont: ", ex);
        }
        return Typeface.DEFAULT;
    }
}

File: app/src/main/java/ru/ulstu_team/ulstuschedule/util/FontUtil.kt
package ru.ulstu_team.ulstuschedule.util

import android.content.Context
import android.graphics.Typeface
import android.util.Log

object FontUtil {

    private val TAG = "FontUtil"

    private fun getFont(context: Context, fontName: String): Typeface {
        var font = fontName
        try {
            val pathBegin = "fonts/"
            if (!font.startsWith(pathBegin))
                font = pathBegin + font
            return Typeface.createFromAsset(context.assets, font)
        } catch (ex: Exception) {
            Log.e(TAG, "getFont: ", ex)
        }

        return Typeface.DEFAULT
    }

    fun getMedium(context: Context) = getFont(context, "Roboto-Medium.ttf")
}
